### PR TITLE
Add UTransport based L2 Notification API implementation

### PR DESCRIPTION
**Pull request by sophokles73**
This is for #122

### CODE REVIEW

**Review by AnotherDaniel — APPROVED** [review]
LGTM

File: src/communication/default_notifier.rs [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+            .map_err(|e| NotificationError::InvalidArgument(e.to_string()))?;
+        self.transport
+            .send(msg)
+            .await
+            .map_err(NotificationError::NotifyError)
+    }
+
+    async fn start_listening(
+        &self,
+        origin_filter: &UUri,
```
Comment (evshary) [inline src/communication/default_notifier.rs:63]: We're using `topic` in the trait. Perhaps we can have a consistent name?

File: src/communication/default_notifier.rs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    pub fn new(transport: Arc<dyn UTransport>, uri_provider: Arc<dyn LocalUriProvider>) -> Self {
+        SimpleNotifier {
+            transport,
+            uri_provider,
+        }
+    }
+}
+
+#[async_trait]
+impl Notifier for SimpleNotifier {
```
Comment (evshary) [inline src/communication/default_notifier.rs:40]: Should we separate `notify` and `start_listening` & `stop_listening` to different traits? If I understand correctly, `notify` is only used by the sender, while the other two are used by the receiver.

**Review by evshary — APPROVED** [review]
LGTM overall. I left some comments

File: src/communication/default_notifier.rs (hunk 2 above)
Comment (sophokles73) [reply src/communication/default_notifier.rs:40]: I have taken inspiration from the [up-java implementation](https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/communication/Notifier.java) where everything is in a single interface. Do you think it has a lot of value to have separate traits here?

Comment (evshary) [reply src/communication/default_notifier.rs:40]: Hmm... Having different traits is clearer, but I prefer to have a more consistent API within different languages. So, it makes sense to me to have the same way as up-java does.

**Pull request merged**